tests/base/polynomials_integrated_legendre_sz.cc

```diff
@@ -0,0 +1,87 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// This tests the stability of the polynomial evaluation of IntegratedLegendreSZ.
+
+#include "../tests.h"
+#include <iomanip>
+#include <fstream>
+#include <cmath>
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/polynomials_integrated_legendre_sz.h>
+#include <deal.II/base/quadrature_lib.h>
+
+
+using namespace Polynomials;
+
+
+void check_at_one (const std::vector<Polynomial<double> > &p)
+{
+  // Ignore first two polynomials as the integrated Legendre polynomials are only defined
+  // for degree > 1, it is only added to maintain the recursive relation.
+
+  deallog << "Function value of polynomial at right end point: ";
+  for (unsigned int i=2; i<p.size(); ++i)
+    {
+      deallog << '.';
+      const double y = p[i].value(1.);
+      if (std::fabs(y) > 1e-13)
+        deallog << "Error1  lg y=" << std::log10(std::fabs(y)) << std::endl;
```

Comment:
Before I switched the order of the division by k between lines 92-110 in polynomials_integrated_legendre_sz.cc I was getting a stability problem after about order ~15, but delaying the division seemed to improve things.

Another option is to switch to the form L_{n} = \frac{l_{n} - l_{n-2}}{2n-1)}, where the lowercase l_{n} is the Legendre polynomial of order n.